provider/oci/storage_volumes.go

```diff
@@ -542,8 +542,6 @@ func (v *volumeSource) getAttachmentStatus(resourceID *string) (string, error) {
 }
 
 func (v *volumeSource) AttachVolumes(ctx envcontext.ProviderCallContext, params []storage.VolumeAttachmentParams) ([]storage.AttachVolumesResult, error) {
-	var credErr error
```

Comment:
Removing this code is incorrect.
We need to record a credential error first up, but then still look through the volumes to record the error in the result struct

Reply:
That's fine, but the old code didn't do that either.

Reply:
@wallyworld I've re-written this code to handle what I believe is a better error handling. It exits out early and has the advantage of now reporting per attachment volume error. Previously if 0 error'd out, subsequent items would also error out, so it wasn't per error..